### PR TITLE
Migrate MUTime

### PR DESCRIPTION
**Pull request by MoveUpwardsDev**

### CODE REVIEW

File: Sejima/Source/MUTime/MUTime.swift [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+/// Class that provide a circular indicator to represent time with customizable options.
+@IBDesignable
+open class MUTime: MUNibView {
+
+    @IBOutlet private var timeLabel: MULabelCounter!
+    @IBOutlet private var timeBackground: UIView!
+
+    /// The progress value text font.
+    @objc open dynamic var timeFont: UIFont = .systemFont(ofSize: 13, weight: .bold) {
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift]: MUTime object will be named close to time so time.timeFont looks like a repetition.

File: Sejima/Source/MUTime/MUTime.swift [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        didSet {
+            timeLabel.textFont = timeFont
+        }
+    }
+
+    /// The progress value text color.
+    @IBInspectable open dynamic var timeColor: UIColor = .clear {
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift]: Idem as timeFont.

File: Sejima/Source/MUTime/MUTime.swift [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        didSet {
+            timeLabel.textColor = timeColor
+        }
+    }
+
+    /// The progress value text background color.
+    @IBInspectable open dynamic var timeBackgroundColor: UIColor = .clear {
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift:40]: Can be an override of backgroundColor and the superview will be always .clear and we just change timeBackground UIView?

File: Sejima/Source/MUTime/MUTime.swift [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    /// Specifies the progress text value format.
+    @IBInspectable open dynamic var format: String = "%.f" {
+        didSet {
+            timeLabel.format = format
+            updateTime()
+        }
+    }
+
+    /// Show / Hide the progress text value. Shown by default.
+    @IBInspectable open dynamic var showValue: Bool = true {
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift:55]: Maybe hidden the timeLabel will be set the format to ""? Or more readable like that?

File: Sejima/Source/MUTime/MUTime.swift [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    // MARK: - Indicator
+
+    /// Define the progress line color.
+    @IBInspectable open dynamic var indicatorColor: UIColor = .clear {
+        didSet {
+            progressPathLayer.strokeColor = indicatorColor.cgColor
+        }
+    }
+
+    /// Specifies the start angle for the progress line path.
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift]: Add that the value is in degrees.

File: Sejima/Source/MUTime/MUTime.swift [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    @IBInspectable open dynamic var indicatorStartAngle: CGFloat = 0 {
+        didSet {
+            setNeedsLayout()
+        }
+    }
+
+    /// Specifies the end angle for the progress line path.
+    @IBInspectable open dynamic var indicatorEndAngle: CGFloat = 360 {
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift]: Idem, and to avoid a cast, 0.0 and 360.0 will be fine.

File: Sejima/Source/MUTime/MUTime.swift [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+            progressPathLayer.lineCap = indicatorLineCap
+        }
+    }
+
+    // MARK: - Animation
+
+    /// Specifies the animation duration for the progress line to go from current progress value up to target value.
+    @objc open dynamic var animationDuration: TimeInterval = 0.3
+
+    // MARK: - Private functions
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift:121]: All this MARK should be right after IBOutlet to be more visible?

File: Sejima/Source/MUTime/MUTime.swift [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    }
+
+    private func progressPath() -> UIBezierPath {
+        let ovalPath = UIBezierPath()
+        let ovalRect = progressFrame()
+        let startAngle = indicatorStartAngle
+        let endAngle = indicatorEndAngle
+        let minValue = indicatorMinValue
+        let maxValue = indicatorMaxValue * 0.5
+        let angleDiff: CGFloat = (startAngle > endAngle) ?
```
Comment (damien-nd) [inline Sejima/Source/MUTime/MUTime.swift]: Useless parenthesis (3 times).

File: Sejima/Source/MUTime/MUTime.swift (hunk 4 above)
Comment (MoveUpwardsDev) [reply Sejima/Source/MUTime/MUTime.swift:55]: C'est vrai mais cela permet de faire un bind avec Rx sur un boolean et c'est plus compréhensible je trouve de faire un hidden que de mettre un format à ""

**Pull request merged**